lib/publiccloud/instance.pm

```diff
@@ -85,6 +86,43 @@ sub run_ssh_command {
     }
 }
 
+=head2 ssh_script_retry
+
+    ssh_script_retry(command[, retry => 3][, delay => 10][, timeout => 90][, ssh_opts =>'..'][, username => 'XXX'][, no_quote => 0]);
+
+Run a C<command> via ssh in the given PC instance until it succeeds or
+the given number of retries is exhausted and an exception is thrown.
+Timeout can be set by C<timeout> or 90 sec by default.
+By default, the command is passed in single quotes to SSH.
+To avoid quoting use C<<no_quote=>1>>.
+With C<<ssh_opts=>'...'>> you can overwrite all default ops which are in
+C<<$instance->ssh_opts>>.
+Use argument C<username> to specify a different username then
+C<<$instance->username()>>.
+=cut
+sub ssh_script_retry {
+    my ($self, $cmd, %args) = @_;
+    $cmd = $args{cmd} if (!defined $cmd && $args{cmd});
+    $args{rc_only} = 1;
```

Comment:
Whys is this in here? It's not used elsewhere - just defined in here.

Reply:
it is used here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11199/files#diff-967d56f05585b47be10c06c83e8554574ce9d259d95bac01abff8d39a7a169a2L57

but yes, myaybe the caller should set `rc_only` when calling `retry_ssh_command()`

Reply:
`rc_only` is obligatory, otherwise there is no possibility to check, if the command was successful. And this is necessary for checking, if we need to run the command again.
I would argue, that for `retry_ssh_command` this is valid because there is no way of realizing this otherwise (while sticking to the current internals).

Reply:
Oh, sorry. I'd maybe make a comment about it..